# Telegram Bot Cursor Rules

**Version:** 3.3.0 ‚≠ê  
**Last Updated:** 13 –æ–∫—Ç—è–±—Ä—è 2025  
**Changelog:** –°–º. [CHANGELOG.mdc](./CHANGELOG.mdc)  
**Lines:** Optimized to < 550 per module (target: 500)

## üìã –û —Å–∏—Å—Ç–µ–º–µ –ø—Ä–∞–≤–∏–ª

–ú–æ–¥—É–ª—å–Ω–∞—è —Å–∏—Å—Ç–µ–º–∞ Cursor Rules –¥–ª—è –ø—Ä–æ–µ–∫—Ç–∞ **Telegram Channel Parser Bot**.

**Aligned with official best practices:**
- ‚úÖ [Cursor Official Docs](https://docs.cursor.com/context/rules)
- ‚úÖ [Trigger.dev Guide](https://trigger.dev/blog/cursor-rules)
- ‚úÖ [awesome-cursorrules](https://github.com/PatrickJS/awesome-cursorrules)

**Features:**
- QR-–∞–≤—Ç–æ—Ä–∏–∑–∞—Ü–∏—è (–±–µ–∑ SMS)
- Admin Panel —á–µ—Ä–µ–∑ Mini App
- RAG-—Å–∏—Å—Ç–µ–º–∞ –¥–ª—è –ø–æ–∏—Å–∫–∞
- –ú–∏–∫—Ä–æ—Å–µ—Ä–≤–∏—Å–Ω–∞—è –∞—Ä—Ö–∏—Ç–µ–∫—Ç—É—Ä–∞

## üóÇÔ∏è –°—Ç—Ä—É–∫—Ç—É—Ä–∞ –º–æ–¥—É–ª–µ–π

–ü—Ä–∞–≤–∏–ª–∞ —Ä–∞–∑–¥–µ–ª–µ–Ω—ã –Ω–∞ **10 –ª–æ–≥–∏—á–µ—Å–∫–∏—Ö –º–æ–¥—É–ª–µ–π** —Å–æ–≥–ª–∞—Å–Ω–æ [Cursor Best Practices](https://docs.cursor.com/context/rules) (< 550 —Å—Ç—Ä–æ–∫ –∫–∞–∂–¥—ã–π):

### Core Modules

| –ú–æ–¥—É–ª—å | –û–ø–∏—Å–∞–Ω–∏–µ | –°—Ç—Ä–æ–∫ | Priority | Rule Type | Scope |
|--------|----------|-------|----------|-----------|-------|
| **01-core.mdc** | –û—Å–Ω–æ–≤–Ω—ã–µ –ø—Ä–∞–≤–∏–ª–∞, –∫—Ä–∏—Ç–∏—á–Ω—ã–µ –ø–∞—Ç—Ç–µ—Ä–Ω—ã | 406 | üî¥ Critical | autoAttached | `telethon/**/*.py` |
| **02-architecture.mdc** | –ú–∏–∫—Ä–æ—Å–µ—Ä–≤–∏—Å—ã, Docker, networking | 332 | üü° Medium | autoAttached | `telethon/**/*.py` |
| **03-database.mdc** | PostgreSQL, Redis, –º–æ–¥–µ–ª–∏, timezone | 433 | üî¥ Critical | autoAttached | `telethon/models.py`, `database.py` |
| **04-development.mdc** | Workflow, testing, dev.sh helper | 487 | üü¢ Low | autoAttached | `telethon/scripts/**` |

### Feature Modules

| –ú–æ–¥—É–ª—å | –û–ø–∏—Å–∞–Ω–∏–µ | –°—Ç—Ä–æ–∫ | Priority | Rule Type | Scope |
|--------|----------|-------|----------|-----------|-------|
| **05-security.mdc** | QR Login, auth, encryption, sessions | 463 | üî¥ High | autoAttached | `telethon/*auth*.py`, `crypto*.py` |
| **06-admin.mdc** | Admin Panel, roles, subscriptions | 537 | üü° Medium | autoAttached | `telethon/admin*.py` |
| **07-rag.mdc** | RAG, vector search, embeddings, AI | 543 | üü° Medium | autoAttached | `telethon/rag_service/**` |
| **08-api.mdc** | FastAPI endpoints, rate limiting | 525 | üü° Medium | autoAttached | `telethon/main.py`, `api_*.py` |
| **09-external.mdc** | External services (Qdrant, Crawl4AI) | 424 | üü¢ Low | autoAttached | `telethon/integrations/**` |
| **10-groups.mdc** | üÜï Groups –¥–∞–π–¥–∂–µ—Å—Ç—ã, —É–ø–æ–º–∏–Ω–∞–Ω–∏—è, n8n | 480 | üü° Medium | autoAttached | `telethon/*group*.py`, `n8n/workflows/group*.json` |

**Special Files:**

| –§–∞–π–ª | –ù–∞–∑–Ω–∞—á–µ–Ω–∏–µ | Priority |
|------|------------|----------|
| **CHANGELOG.mdc** | –ò—Å—Ç–æ—Ä–∏—è –≤–µ—Ä—Å–∏–π –∏ –∏–∑–º–µ–Ω–µ–Ω–∏–π | üü¢ Low |
| **README.md** | –ù–∞–≤–∏–≥–∞—Ü–∏—è –∏ –¥–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—è —Å–∏—Å—Ç–µ–º—ã | üü¢ Low |

**Entry Point (–≤ –∫–æ—Ä–Ω–µ –ø—Ä–æ–µ–∫—Ç–∞):**

| –§–∞–π–ª | –ù–∞–∑–Ω–∞—á–µ–Ω–∏–µ | Priority |
|------|------------|----------|
| **`.cursorrules`** | Entry point, critical rules, verification checklist | üî¥ Critical |

## üéØ –ö–∞–∫ –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å

### –û—Å–Ω–æ–≤–Ω–æ–π workflow

1. **Entry point:** `.cursorrules` –≤ –∫–æ—Ä–Ω–µ –ø—Ä–æ–µ–∫—Ç–∞ ‚Äî –∑–∞–≥—Ä—É–∂–∞–µ—Ç—Å—è –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏
2. **Cursor –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏ –∑–∞–≥—Ä—É–∑–∏—Ç –º–æ–¥—É–ª–∏** –ø–æ `scope` (file patterns)
3. **–î–ª—è –∫–æ–Ω–∫—Ä–µ—Ç–Ω—ã—Ö –∑–∞–¥–∞—á** –∏—Å–ø–æ–ª—å–∑—É–π—Ç–µ —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É—é—â–∏–π –º–æ–¥—É–ª—å –∏–∑ `.cursor/rules/telegram-bot/`

### Cursor Settings

**–†–µ–∫–æ–º–µ–Ω–¥—É–µ–º—ã–µ –Ω–∞—Å—Ç—Ä–æ–π–∫–∏ –¥–ª—è `.cursor/settings.json`:**

```json
{
  "cursor.rules.autoAttach": true,
  "cursor.rules.scope": "workspace",
  "cursor.rules.maxFiles": 3,
  "cursor.rules.priority": "user"
}
```

### Auto-Attach –ø–æ Scope

–ú–æ–¥—É–ª–∏ –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏ –ø—Ä–∏–º–µ–Ω—è—é—Ç—Å—è –ø—Ä–∏ —Ä–µ–¥–∞–∫—Ç–∏—Ä–æ–≤–∞–Ω–∏–∏ —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É—é—â–∏—Ö —Ñ–∞–π–ª–æ–≤:

```yaml
# 01-core.mdc
scope:
  - "telethon/**/*.py"
  - "!telethon/tests/**"

# 03-database.mdc
scope:
  - "telethon/models.py"
  - "telethon/database.py"
  - "telethon/migrations/**"

# 05-security.mdc
scope:
  - "telethon/*auth*.py"
  - "telethon/crypto*.py"
  - "telethon/sessions/**"
```

## üìö Quick Links

### –ü–æ –∑–∞–¥–∞—á–∞–º

**üÜï –°–æ–∑–¥–∞–Ω–∏–µ –Ω–æ–≤–æ–≥–æ –∫–æ–º–ø–æ–Ω–µ–Ω—Ç–∞:**
1. `telegram-bot.cursorrules` ‚Üí critical rules
2. `01-core.mdc` ‚Üí —Å—Ç—Ä—É–∫—Ç—É—Ä–∞ –ø—Ä–æ–µ–∫—Ç–∞
3. `02-architecture.mdc` ‚Üí –≥–¥–µ —Ä–∞–∑–º–µ—Å—Ç–∏—Ç—å –∫–æ–¥
4. `03-database.mdc` ‚Üí –º–æ–¥–µ–ª–∏ –ë–î (–µ—Å–ª–∏ –Ω—É–∂–Ω—ã)
5. `08-api.mdc` ‚Üí API endpoints (–µ—Å–ª–∏ –Ω—É–∂–Ω—ã)

**üêõ Debugging:**
1. `telegram-bot.cursorrules` ‚Üí verification checklist
2. `04-development.mdc` ‚Üí –ª–æ–≥–∏, –æ—Ç–ª–∞–¥–∫–∞
3. –°–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É—é—â–∏–π feature module

**üîê –†–∞–±–æ—Ç–∞ —Å authentication:**
1. `05-security.mdc` ‚Üí QR Login, sessions
2. `03-database.mdc` ‚Üí User model, Redis
3. `telegram-bot.cursorrules` ‚Üí critical patterns

**üëë Admin —Ñ—É–Ω–∫—Ü–∏–∏:**
1. `06-admin.mdc` ‚Üí Admin Panel, roles
2. `03-database.mdc` ‚Üí SubscriptionHistory
3. `08-api.mdc` ‚Üí Admin API endpoints

**ü§ñ RAG/AI —Ñ—É–Ω–∫—Ü–∏–∏:**
1. `07-rag.mdc` ‚Üí RAG pipeline, embeddings
2. `09-external.mdc` ‚Üí Qdrant, GigaChat
3. `03-database.mdc` ‚Üí IndexingStatus

## ‚ö†Ô∏è –ö—Ä–∏—Ç–∏—á–Ω—ã–µ –ø—Ä–∞–≤–∏–ª–∞ (must-read)

### –ò–∑ `telegram-bot.cursorrules` –∏ `01-core.mdc`

‚úÖ **#0: –í–°–ï–ì–î–ê –∏—Å–ø–æ–ª—å–∑—É–π—Ç–µ Context7 –ø–µ—Ä–µ–¥ —Ä–∞–∑—Ä–∞–±–æ—Ç–∫–æ–π**

```bash
# ‚úÖ –ü–†–ê–í–ò–õ–¨–ù–´–ô WORKFLOW:
# 1. –ò—Å–ø–æ–ª—å–∑—É–π—Ç–µ Context7 –¥–ª—è –ø–æ–∏—Å–∫–∞ —Ä–µ—à–µ–Ω–∏–π
Context7: "Telegram Bot API best practices"
Context7: "FastAPI async patterns"
Context7: "SQLAlchemy relationships PostgreSQL"

# 2. –ò–∑—É—á–∏—Ç–µ –æ—Ñ–∏—Ü–∏–∞–ª—å–Ω—É—é –¥–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—é —á–µ—Ä–µ–∑ Context7
#    - –ù–∞–π–¥–∏—Ç–µ –ø—Ä–∏–º–µ—Ä—ã –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω–∏—è
#    - –ü—Ä–æ–≤–µ—Ä—å—Ç–µ –∞–∫—Ç—É–∞–ª—å–Ω—ã–µ –≤–µ—Ä—Å–∏–∏ API
#    - –£–∑–Ω–∞–π—Ç–µ best practices

# 3. –¢–û–õ–¨–ö–û –ü–û–¢–û–ú –ø–∏—à–∏—Ç–µ –∫–æ–¥
```

**Auto-integration hint –¥–ª—è AI:**
```python
# If uncertain about library usage, automatically query:
# Context7("library-name best practices")
# Context7("library-name current API version")
```

**–ü—Ä–∏–º–µ—Ä—ã:**
```bash
# –ü–µ—Ä–µ–¥ –¥–æ–±–∞–≤–ª–µ–Ω–∏–µ–º Telegram Mini App
‚Üí Context7: "Telegram Mini Apps WebAppInfo best practices"

# –ü–µ—Ä–µ–¥ —Ä–∞–±–æ—Ç–æ–π —Å FastAPI
‚Üí Context7: "FastAPI background tasks async"

# –ü–µ—Ä–µ–¥ –∏–∑–º–µ–Ω–µ–Ω–∏–µ–º SQLAlchemy –º–æ–¥–µ–ª–µ–π
‚Üí Context7: "SQLAlchemy relationships many-to-many"

# –ü–µ—Ä–µ–¥ –∏–Ω—Ç–µ–≥—Ä–∞—Ü–∏–µ–π –≤–Ω–µ—à–Ω–µ–≥–æ API
‚Üí Context7: "Qdrant Python client async"
```

---

‚úÖ **#1: –¢–û–õ–¨–ö–û PostgreSQL** (–ë–ï–ó SQLite fallback)

```python
# ‚úÖ –ü—Ä–∞–≤–∏–ª—å–Ω–æ
database_url = os.getenv("TELEGRAM_DATABASE_URL")
if "sqlite" in database_url.lower():
    raise ValueError("SQLite –ù–ï –ø–æ–¥–¥–µ—Ä–∂–∏–≤–∞–µ—Ç—Å—è!")

# ‚ùå –ù–ò–ö–û–ì–î–ê
db_url = os.getenv("DATABASE_URL", "sqlite:///fallback.db")  # –ù–ï–¢!
```

---

‚úÖ **#2: Timezone-aware datetime –í–°–ï–ì–î–ê**

```python
from datetime import datetime, timezone

# ‚úÖ –ü—Ä–∞–≤–∏–ª—å–Ω–æ
user.created_at = datetime.now(timezone.utc)

# ‚ùå Naive datetime - –ù–ï–¢!
user.created_at = datetime.now()
```

---

‚úÖ **#3: –§–∏–ª—å—Ç—Ä–∞—Ü–∏—è –ø–æ user_id –û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–ê**

```python
# ‚úÖ –ü—Ä–∞–≤–∏–ª—å–Ω–æ
posts = db.query(Post).filter(Post.user_id == user_id).all()

# ‚ùå –£—Ç–µ—á–∫–∞ –¥–∞–Ω–Ω—ã—Ö
posts = db.query(Post).all()
```

---

‚úÖ **#4: Redis –ë–ï–ó –ø–∞—Ä–æ–ª—è + prefixes**

```python
# ‚úÖ –ü—Ä–∞–≤–∏–ª—å–Ω–æ (Valkey default)
redis_client = redis.Redis(host='redis', port=6379)

# ‚úÖ –í—Å–µ–≥–¥–∞ –∏—Å–ø–æ–ª—å–∑—É–π—Ç–µ prefixes
redis.setex(f"qr_session:{session_id}", 300, data)

# ‚ùå –ù–ï–¢ –ø–∞—Ä–æ–ª—è!
redis_client = redis.Redis(host='redis', password='...')  # –ù–ï–¢!

# ‚ùå –ù–ï–¢ prefix - –∫–æ–Ω—Ñ–ª–∏–∫—Ç—ã!
redis.setex(session_id, 300, data)
```

### –ò–∑ `03-database.mdc`

‚úÖ **Timezone handling –≤ PostgreSQL:**

```python
# ‚úÖ –í—Å–µ–≥–¥–∞ UTC –≤ –ë–î, Europe/Moscow –¥–ª—è display
from datetime import datetime, timezone
from zoneinfo import ZoneInfo

# –°–æ—Ö—Ä–∞–Ω—è–µ–º
user.created_at = datetime.now(timezone.utc)

# –ü–æ–∫–∞–∑—ã–≤–∞–µ–º –ø–æ–ª—å–∑–æ–≤–∞—Ç–µ–ª—é
LOCAL_TZ = ZoneInfo('Europe/Moscow')
display_time = user.created_at.astimezone(LOCAL_TZ)

# –°—Ä–∞–≤–Ω–µ–Ω–∏–µ
if expires.tzinfo is None:
    expires = expires.replace(tzinfo=timezone.utc)
```

### –ò–∑ `04-development.mdc`

‚úÖ **Context7 Workflow –≤ development cycle:**

```bash
# 0. –û–ë–Ø–ó–ê–¢–ï–õ–¨–ù–û: –ò–∑—É—á–∏—Ç—å –¥–æ–∫—É–º–µ–Ω—Ç–∞—Ü–∏—é —á–µ—Ä–µ–∑ Context7
# ‚Üí Context7: "Feature name best practices"
# ‚Üí Context7: "Library name API documentation"

# 1. –í–Ω–µ—Å—Ç–∏ –∏–∑–º–µ–Ω–µ–Ω–∏—è
vim telethon/qr_auth_manager.py

# 2. –ü–µ—Ä–µ—Å–æ–±—Ä–∞—Ç—å
telethon-rebuild

# 3. –°–º–æ—Ç—Ä–µ—Ç—å –ª–æ–≥–∏
telethon-logs

# 4. –¢–µ—Å—Ç–∏—Ä–æ–≤–∞—Ç—å
curl http://localhost:8010/docs
```

## üîÄ Conflict Resolution

### Priority System

–ü—Ä–∏ –∫–æ–Ω—Ñ–ª–∏–∫—Ç–µ –ø—Ä–∞–≤–∏–ª –∏–∑ —Ä–∞–∑–Ω—ã—Ö –º–æ–¥—É–ª–µ–π:

1. **01-core.mdc** (HIGHEST PRIORITY)
   - Critical patterns always apply
   - PostgreSQL, timezone, user filtering

2. **03-database.mdc & 05-security.mdc** (HIGH)
   - Database rules for data persistence
   - Security rules for auth logic

3. **07-rag.mdc & 08-api.mdc** (MEDIUM)
   - Feature-specific implementations
   - API endpoint patterns

4. **02-architecture.mdc & 04-development.mdc** (LOW)
   - Architectural guidance
   - Development workflow

### Resolution Strategy

```yaml
# If core rule conflicts with feature rule:
‚Üí follow 01-core.mdc

# If security conflicts with database:
‚Üí follow 05-security.mdc for auth logic
‚Üí follow 03-database.mdc for persistence

# If both database and RAG apply:
‚Üí follow 03-database.mdc for models
‚Üí follow 07-rag.mdc for embeddings/search

# If in doubt:
‚Üí follow higher priority module
‚Üí check telegram-bot.cursorrules
```

## ‚úÖ Verification Checklist

–°–º. –ø–æ–ª–Ω—ã–π checklist –≤ `telegram-bot.cursorrules` —Å–µ–∫—Ü–∏—è "Verification Checklist Summary".

**–ö—Ä–∞—Ç–∫–∞—è –≤–µ—Ä—Å–∏—è:**

- [ ] Context7 –∏—Å–ø–æ–ª—å–∑–æ–≤–∞–Ω –¥–ª—è –∏–∑—É—á–µ–Ω–∏—è API
- [ ] PostgreSQL only (`grep -r "sqlite"` ‚Üí 0 —Ä–µ–∑—É–ª—å—Ç–∞—Ç–æ–≤)
- [ ] Timezone-aware datetime –≤–µ–∑–¥–µ
- [ ] User ID filtering –≤–æ –≤—Å–µ—Ö queries
- [ ] Type hints –Ω–∞ –≤—Å–µ—Ö —Ñ—É–Ω–∫—Ü–∏—è—Ö
- [ ] Async patterns –¥–ª—è HTTP/DB
- [ ] Redis prefixes –¥–ª—è –∏–∑–æ–ª—è—Ü–∏–∏
- [ ] Error handling —Å try-except

## üö® Troubleshooting

**Cursor –Ω–µ –ø—Ä–∏–º–µ–Ω—è–µ—Ç –ø—Ä–∞–≤–∏–ª–∞:**
1. –ü—Ä–æ–≤–µ—Ä—å—Ç–µ `.cursor/rules` –≤ –∫–æ—Ä–Ω–µ workspace
2. –û—Ç–∫—Ä–æ–π—Ç–µ Command Palette ‚Üí "Cursor Rules: Reload"
3. –ü—Ä–æ–≤–µ—Ä—å—Ç–µ —á—Ç–æ —Ñ–∞–π–ª —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É–µ—Ç `scope` pattern
4. –ü—Ä–æ–≤–µ—Ä—å—Ç–µ `autoAttach: true` –≤ YAML frontmatter

**–ü—Ä–∞–≤–∏–ª–∞ –∫–æ–Ω—Ñ–ª–∏–∫—Ç—É—é—Ç:**
1. –°–º–æ—Ç—Ä–∏—Ç–µ —Å–µ–∫—Ü–∏—é "Conflict Resolution" –≤—ã—à–µ
2. `01-core.mdc` –∏–º–µ–µ—Ç highest priority
3. –ü—Ä–∏ –∫–æ–Ω—Ñ–ª–∏–∫—Ç–µ ‚Äî —Å–ª–µ–¥—É–π—Ç–µ `telegram-bot.cursorrules`

**–ù—É–∂–Ω–æ –¥–æ–±–∞–≤–∏—Ç—å –Ω–æ–≤–æ–µ –ø—Ä–∞–≤–∏–ª–æ:**
1. –û–ø—Ä–µ–¥–µ–ª–∏—Ç–µ –∫–∞—Ç–µ–≥–æ—Ä–∏—é (core/database/security/etc.)
2. –î–æ–±–∞–≤—å—Ç–µ –≤ —Å–æ–æ—Ç–≤–µ—Ç—Å—Ç–≤—É—é—â–∏–π –º–æ–¥—É–ª—å
3. –ï—Å–ª–∏ –º–æ–¥—É–ª—å > 600 —Å—Ç—Ä–æ–∫ ‚Üí —Ä–∞–∑–¥–µ–ª–∏—Ç–µ –Ω–∞ –ø–æ–¥–º–æ–¥—É–ª–∏
4. –û–±–Ω–æ–≤–∏—Ç–µ YAML frontmatter (`scope`, `tags`, `priority`)
5. –î–æ–±–∞–≤—å—Ç–µ quick examples –≤ –∫–æ–Ω—Ü–µ –º–æ–¥—É–ª—è
6. –û–±–Ω–æ–≤–∏—Ç–µ CHANGELOG.mdc
7. –û–±–Ω–æ–≤–∏—Ç–µ —ç—Ç–æ—Ç README

**AI –Ω–µ –∏—Å–ø–æ–ª—å–∑—É–µ—Ç Context7:**
1. –ü—Ä–æ–≤–µ—Ä—å—Ç–µ —á—Ç–æ –ø—Ä–∞–≤–∏–ª–æ #0 –≤ `telegram-bot.cursorrules`
2. –î–æ–±–∞–≤—å—Ç–µ auto-integration hint –≤ –º–æ–¥—É–ª—å
3. –Ø–≤–Ω–æ –Ω–∞–ø–æ–º–Ω–∏—Ç–µ AI –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å Context7

## üìä –°—Ç–∞—Ç–∏—Å—Ç–∏–∫–∞

**–§–∞–π–ª–æ–≤–∞—è —Å—Ç—Ä—É–∫—Ç—É—Ä–∞:**
- Total modules: **10** (+ Groups module üÜï)
- Total lines: ~4,630 (–æ–ø—Ç–∏–º–∏–∑–∏—Ä–æ–≤–∞–Ω–æ —Å 4,435, +480 –¥–ª—è Groups)
- Average per module: ~463 lines (target: < 500)
- Files with YAML frontmatter: 100%
- Rule Type specified: 100%

**–û–ø—Ç–∏–º–∏–∑–∞—Ü–∏—è (v3.3.0):**
- –°–æ–∫—Ä–∞—â–µ–Ω–æ –Ω–∞ 286 —Å—Ç—Ä–æ–∫ (v3.2 ‚Üí v3.3)
- **–î–æ–±–∞–≤–ª–µ–Ω –º–æ–¥—É–ª—å Groups** (+480 —Å—Ç—Ä–æ–∫ –¥–ª—è –Ω–æ–≤–æ–≥–æ —Ñ—É–Ω–∫—Ü–∏–æ–Ω–∞–ª–∞)
- 5 —Ñ–∞–π–ª–æ–≤ –æ–ø—Ç–∏–º–∏–∑–∏—Ä–æ–≤–∞–Ω—ã –¥–æ < 550 —Å—Ç—Ä–æ–∫
- –í—Å–µ –º–æ–¥—É–ª–∏ —Å–ª–µ–¥—É—é—Ç Cursor best practices

**–ü–æ–∫—Ä—ã—Ç–∏–µ scope:**
- Core files: `telethon/**/*.py` (excluding tests/scripts)
- Database: `telethon/models.py`, `telethon/database.py`
- Security: `telethon/*auth*.py`, `telethon/crypto*.py`
- RAG: `telethon/rag_service/**`
- API: `telethon/main.py`, `telethon/api_*.py`

## üîÑ Updates

–°–º. [CHANGELOG.mdc](./CHANGELOG.mdc) –¥–ª—è –¥–µ—Ç–∞–ª—å–Ω–æ–π –∏—Å—Ç–æ—Ä–∏–∏ –∏–∑–º–µ–Ω–µ–Ω–∏–π.

**Latest (v3.3.0 - 2025-10-13):**
- ‚úÖ **Optimized to < 550 lines** per module (best practices)
- ‚úÖ **Rule Type** metadata (`always`, `autoAttached`, `manual`)
- ‚úÖ **Structured sections** following [Trigger.dev guide](https://trigger.dev/blog/cursor-rules)
- ‚úÖ **Enhanced Quick Examples** (‚úÖ Correct / ‚ùå Bad patterns)
- ‚úÖ **Testing section** in main `.cursorrules`
- ‚úÖ **Common Pitfalls** tables with solutions
- ‚úÖ **Priority system** in YAML frontmatter
- ‚úÖ **Better Verification Checklists** per module
- ‚úÖ Reduced total lines by 286 (4,435 ‚Üí 4,150)

## üìû Support & Resources

**Project:**
- **GitHub Issues:** [telethon/issues](https://github.com/your-repo/telethon/issues)
- **Documentation:** `/home/ilyasni/n8n-server/n8n-installer/telethon/docs/`
- **Changelog:** [CHANGELOG.mdc](./CHANGELOG.mdc)

**Cursor Rules Best Practices:**
- üìñ [Cursor Official Docs](https://docs.cursor.com/context/rules) - Rule types, scope, auto-attach
- üéØ [Trigger.dev Guide](https://trigger.dev/blog/cursor-rules) - How to write great rules (10 tips)
- ‚≠ê [awesome-cursorrules](https://github.com/PatrickJS/awesome-cursorrules) - Community examples
- üí¨ [Cursor Forum - Best Practices](https://forum.cursor.com/t/my-best-practices-for-mdc-rules-and-troubleshooting/50526)
- üí¨ [Reddit Discussion](https://www.reddit.com/r/cursor/comments/1jhurjt/best_practices_for_cursor_rules/)

**Key Recommendations:**
- ‚úÖ Keep rules under 500 lines ([Cursor Docs](https://docs.cursor.com/context/rules))
- ‚úÖ Divide into separate modules if growing large
- ‚úÖ Use proper Rule Types (always, autoAttached, manual)
- ‚úÖ Provide good/bad examples side-by-side
- ‚úÖ Include verification steps and testing

---

**Maintained by:** Telegram Bot Team  
**License:** CC0-1.0  
**Version:** 3.3.0 ‚≠ê  
**Optimized:** Following official Cursor best practices

